tools/OpenConsole.psm1

```diff
@@ -363,16 +363,91 @@ function Invoke-ClangFormat {
     }
 }
 
+function Invoke-StripBOM {
+    [CmdletBinding()]
+    Param (
+        [Parameter(Mandatory=$true,ValueFromPipeline=$true)]
+        [string]$Path
+    )
+
+    $files = Get-ChildItem -Path "$Path\*" -Include *.xaml
+    foreach ($file in $files) {
+        $content = Get-Content $file
+        [IO.File]::WriteAllLines($file.FullName, $content)
+    }
+}
+
+
 #.SYNOPSIS
-# runs code formatting on all c++ files
+# Check that xaml files are formatted correctly. This won't actually
+# format the files - it'll only ensure that they're formatted correctly.
+function Invoke-VerifyXamlFormat() {
+    $root = Find-OpenConsoleRoot
+    & dotnet tool restore --add-source https://api.nuget.org/v3/index.json
+
+    dotnet tool run xstyler -- -c "$root\Settings.XamlStyler" -d "$root\src\cascadia\TerminalApp" --passive
+    if ($lastExitCode -eq 1) {
+        throw "Xaml formatting bad, run Invoke-XamlFormat on branch"
+    }
+
+    dotnet tool run xstyler -- -c "$root\Settings.XamlStyler" -d "$root\src\cascadia\TerminalControl" --passive
+    if ($lastExitCode -eq 1) {
+        throw "Xaml formatting bad, run Invoke-XamlFormat on branch"
+    }
+
+    dotnet tool run xstyler -- -c "$root\Settings.XamlStyler" -d "$root\src\cascadia\TerminalSettingsEditor" --passive
+    if ($lastExitCode -eq 1) {
+        throw "Xaml formatting bad, run Invoke-XamlFormat on branch"
+    }
+
+}
+
+#.SYNOPSIS
+# run xstyler on xaml files. Note that this will `touch` every file,

```

Comment:
why's that? Is it because of xstyler, or is it because of stripbom?

Reply:
It's because of `xstyler` I'm 99% sure. I think it will write the file even when nothing has changed, which differs form clang_format (which will only write when a file actually changes)